#  Data Understanding: Patient Medical History Features (Part 2)

## Notebook Purpose

This notebook continues the exploration of **patient medical history variables (`hx_` columns)** used in the **Triagegeist emergency triage prediction task**.

Medical history plays a critical role in emergency medicine because **pre-existing conditions influence patient vulnerability, disease severity, and triage prioritization**. Understanding these features helps build models that better reflect **real-world clinical decision-making in emergency departments**.

---

## Features Covered in This Notebook

In this notebook, we continue analyzing additional **medical history (`hx_`) columns** by examining the **value counts** for each feature.

These variables indicate whether a patient has previously been diagnosed with specific medical conditions.

Each feature is encoded as a **binary indicator**:

- **0** → No known history of the condition  
- **1** → Patient has a history of the condition

The following conditions are explored in this notebook:

- `hx_epilepsy`
- `hx_hypothyroidism`
- `hx_hyperthyroidism`
- `hx_hiv`
- `hx_coagulopathy`
- `hx_immunosuppressed`
- `hx_pregnant`
- `hx_substance_use_disorder`
- `hx_coronary_artery_disease`
- `hx_stroke_prior`
- `hx_peripheral_vascular_disease`

---

## Goal of This Notebook

The goal of this notebook is to:

- Examine the **distribution of each medical history feature**
- Understand the **prevalence of different health conditions** in the dataset
- Verify the **binary encoding of the variables**
- Gain insight into **patient comorbidities that may affect emergency triage severity**

This step forms part of the **data understanding phase** before moving to deeper analysis and feature engineering.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as pyplot
import seaborn as sns


In [2]:
df_patient_history = pd.read_csv("../data/patient_history.csv")
df_patient_history.sample(5)

,patient_id,hx_hypertension,hx_diabetes_type2,hx_diabetes_type1,hx_asthma,hx_copd,hx_heart_failure,hx_atrial_fibrillation,hx_ckd,hx_liver_disease,...,hx_hypothyroidism,hx_hyperthyroidism,hx_hiv,hx_coagulopathy,hx_immunosuppressed,hx_pregnant,hx_substance_use_disorder,hx_coronary_artery_disease,hx_stroke_prior,hx_peripheral_vascular_disease
78987,TG-V3TAVWNFL,0,0,1,0,0,1,1,1,0,...,0,0,0,0,0,0,0,0,0,0
43296,TG-B308GP81L,0,1,0,0,0,0,1,0,0,...,0,0,0,0,0,0,0,0,0,0
32777,TG-2T9WQKJT6,0,0,0,1,0,0,1,1,0,...,0,1,0,0,0,0,0,1,0,0
16363,TG-0U5DU0CWK,0,0,0,0,0,0,0,0,1,...,0,0,0,0,0,0,0,1,0,1
88237,TG-DJL4VG67T,0,1,0,0,0,0,0,1,0,...,0,0,0,1,0,0,0,0,1,1


In [4]:
df_patient_history['hx_epilepsy'].value_counts()


hx_epilepsy
0    80200
1    19800
Name: count, dtype: int64

## Column: `hx_epilepsy`

### 1️ What This Column Means

- **hx** = history  
- **epilepsy** = neurological seizure disorder

This column represents:

> **Whether the patient has a prior medical history of epilepsy.**

It indicates that the patient was diagnosed with **epilepsy before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of epilepsy | 80,200 |
| **1** | Has epilepsy history | 19,800 |

Total patients: **100,000**

Approximately:

- **80.2%** → No epilepsy history  
- **19.8%** → Epilepsy history  

---

### 3️ What is Epilepsy

**Epilepsy** is a **chronic neurological disorder** characterized by **recurrent seizures** caused by abnormal electrical activity in the brain.

Seizures may cause:

- loss of consciousness  
- uncontrolled muscle movements  
- confusion or disorientation  
- temporary behavioral changes  

Triggers for seizures can include:

- stress  
- sleep deprivation  
- infections  
- medication changes  
- metabolic disturbances  

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients with epilepsy may present to the emergency department due to:

- **active seizures**
- **status epilepticus** (continuous seizures)
- **post-seizure confusion**
- **injuries caused during seizures**

Status epilepticus is a **life-threatening neurological emergency** that requires **immediate medical treatment**.

Additionally, seizures may also indicate **serious underlying problems** such as:

- brain infection  
- head injury  
- metabolic imbalance  

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider epilepsy history when evaluating symptoms such as:

- loss of consciousness  
- abnormal movements  
- confusion  
- sudden behavioral changes  

Example scenario:

Patient arrives with:

- confusion  
- weakness  
- altered mental state  

If **hx_epilepsy = 1**, clinicians may suspect:

- **recent seizure activity**
- **post-ictal state (recovery phase after a seizure)**
- **neurological instability**

These cases may require **urgent neurological evaluation**.

---


This helps the model identify **patients at risk of neurological emergencies**.

---

### 6 Feature Type

`hx_epilepsy` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Neurological risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature provides information about **baseline neurological vulnerability**.

Patients with epilepsy are more likely to experience **seizures or neurological complications**, which may significantly affect emergency severity assessment.

Therefore, this variable helps the model better identify **patients at risk of seizure-related emergencies**.


In [6]:
df_patient_history['hx_hypothyroidism'].value_counts()


hx_hypothyroidism
0    80266
1    19734
Name: count, dtype: int64

## Column: `hx_hypothyroidism`

### 1️ What This Column Means

- **hx** = history  
- **hypothyroidism** = underactive thyroid condition

This column represents:

> **Whether the patient has a prior medical history of hypothyroidism.**

It indicates that the patient was diagnosed with **hypothyroidism before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of hypothyroidism | 80,266 |
| **1** | Has hypothyroidism history | 19,734 |

Total patients: **100,000**

Approximately:

- **80.3%** → No hypothyroidism history  
- **19.7%** → Hypothyroidism history  

---

### 3️ What is Hypothyroidism

**Hypothyroidism** is a condition where the **thyroid gland does not produce enough thyroid hormones**.

The thyroid hormones regulate important body functions such as:

- metabolism  
- heart rate  
- body temperature  
- energy levels  

Common causes include:

- autoimmune disease (Hashimoto’s thyroiditis)  
- thyroid surgery  
- iodine deficiency  
- certain medications  

Symptoms may include:

- fatigue  
- weight gain  
- cold intolerance  
- slow heart rate  
- depression  
- dry skin  

---

### 4️ Why This Feature Is Important in Emergency Triage

Although hypothyroidism is usually a **chronic manageable condition**, severe cases can lead to serious complications such as:

- **myxedema coma** (a rare but life-threatening condition)
- severe fatigue and mental slowing
- cardiovascular effects such as **slow heart rate**
- metabolic disturbances

Patients with untreated or poorly managed hypothyroidism may experience **reduced physiological resilience during acute illness**.

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider hypothyroidism history when patients present with symptoms such as:

- extreme fatigue  
- confusion  
- slow heart rate  
- hypothermia  

Example scenario:

Patient arrives with:

- confusion  
- low body temperature  
- low heart rate  

If **hx_hypothyroidism = 1**, clinicians may suspect:

- **severe hypothyroid complications**
- metabolic imbalance
- endocrine-related illness

These situations may require **urgent medical assessment and hormone management**.

---


This helps the model capture **endocrine-related health vulnerability**.

---

### 6 Feature Type

`hx_hypothyroidism` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Endocrine risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature helps represent **baseline endocrine system health**.

Patients with hypothyroidism may experience **metabolic and cardiovascular complications during illness**, which can influence the severity of emergency presentations.

Therefore, this variable contributes to improving the model’s ability to **identify patients with underlying endocrine vulnerabilities**.


In [8]:
df_patient_history['hx_hyperthyroidism'].value_counts()


hx_hyperthyroidism
0    80077
1    19923
Name: count, dtype: int64

## Column: `hx_hyperthyroidism`

### 1️ What This Column Means

- **hx** = history  
- **hyperthyroidism** = overactive thyroid condition

This column represents:

> **Whether the patient has a prior medical history of hyperthyroidism.**

It indicates that the patient was diagnosed with **hyperthyroidism before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of hyperthyroidism | 80,077 |
| **1** | Has hyperthyroidism history | 19,923 |

Total patients: **100,000**

Approximately:

- **80.1%** → No hyperthyroidism history  
- **19.9%** → Hyperthyroidism history  

---

### 3️ What is Hyperthyroidism

**Hyperthyroidism** is a condition where the **thyroid gland produces excessive thyroid hormones**, causing the body's metabolism to become overactive.

Thyroid hormones control:

- metabolism  
- heart rate  
- body temperature  
- energy usage  

When hormone levels become too high, body systems may become **overstimulated**.

Common causes include:

- **Graves’ disease** (autoimmune disorder)  
- thyroid nodules  
- thyroid inflammation  

Common symptoms include:

- rapid heart rate  
- weight loss  
- excessive sweating  
- anxiety or nervousness  
- tremors  
- heat intolerance  

---

### 4️ Why This Feature Is Important in Emergency Triage

Hyperthyroidism can lead to serious complications that may require emergency care, including:

- **thyroid storm** (a life-threatening condition)
- severe tachycardia (very fast heart rate)
- cardiac arrhythmias
- heart failure
- metabolic instability

**Thyroid storm** is a rare but extremely dangerous emergency where thyroid hormone levels become dangerously high.

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider hyperthyroidism history when patients present with symptoms such as:

- rapid heart rate  
- palpitations  
- fever  
- agitation or anxiety  
- unexplained weight loss  

Example scenario:

Patient arrives with:

- extremely high heart rate  
- fever  
- confusion  

If **hx_hyperthyroidism = 1**, clinicians may suspect:

- **thyroid storm**
- severe endocrine imbalance
- cardiac complications related to thyroid disease

These conditions may require **urgent medical stabilization**.

---


This allows the model to recognize **patients with endocrine-related cardiovascular risk**.

---

### 6 Feature Type

`hx_hyperthyroidism` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Endocrine risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature represents **baseline endocrine system vulnerability**.

Patients with hyperthyroidism may experience **cardiovascular stress, metabolic imbalance, and endocrine emergencies**, which can increase the likelihood of severe outcomes during an emergency visit.

Therefore, this variable helps the model better predict **patients at risk of serious endocrine-related emergencies**.


In [10]:
df_patient_history['hx_hiv'].value_counts()


hx_hiv
0    80466
1    19534
Name: count, dtype: int64

## Column: `hx_hiv`

### 1️ What This Column Means

- **hx** = history  
- **HIV** = Human Immunodeficiency Virus infection

This column represents:

> **Whether the patient has a prior medical history of HIV infection.**

It indicates that the patient has previously been diagnosed with **HIV before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of HIV | 80,466 |
| **1** | Has HIV history | 19,534 |

Total patients: **100,000**

Approximately:

- **80.5%** → No HIV history  
- **19.5%** → HIV history  

---

### 3️ What is HIV

**Human Immunodeficiency Virus (HIV)** is a virus that attacks the **immune system**, specifically the **CD4 T-cells**, which are responsible for helping the body fight infections.

If untreated, HIV can progress to **Acquired Immunodeficiency Syndrome (AIDS)**.

HIV weakens the immune system, making patients more vulnerable to:

- infections  
- certain cancers  
- opportunistic diseases  

With modern treatments such as **antiretroviral therapy (ART)**, many people with HIV can live long and healthy lives.

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients with HIV may be **immunocompromised**, meaning their immune systems may not respond effectively to infections.

This increases the risk of:

- **severe infections**
- **opportunistic infections**
- **rapid disease progression**
- **systemic complications**

Common emergency conditions in HIV patients include:

- pneumonia  
- severe bacterial infections  
- neurological infections  
- systemic inflammatory responses  

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider HIV history when evaluating symptoms such as:

- fever  
- severe infections  
- unexplained weakness  
- respiratory distress  
- neurological symptoms  

Example scenario:

Patient arrives with:

- high fever  
- difficulty breathing  
- fatigue  

If **hx_hiv = 1**, clinicians may suspect:

- **opportunistic infection**
- **immune-related complications**
- **severe systemic infection**

These situations may require **urgent diagnostic evaluation and treatment**.

---


This helps the model recognize **patients with weakened immune defenses**.

---

### 6 Feature Type

`hx_hiv` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Immune system risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature provides information about **immune system vulnerability**.

Patients with HIV may be more susceptible to **serious infections and systemic complications**, which can significantly influence emergency severity assessment.

Therefore, this variable helps the model better identify **patients at risk of infection-related emergencies**.


In [12]:
df_patient_history['hx_coagulopathy'].value_counts()


hx_coagulopathy
0    80206
1    19794
Name: count, dtype: int64

## Column: `hx_coagulopathy`

### 1️ What This Column Means

- **hx** = history  
- **coagulopathy** = disorder affecting blood clotting

This column represents:

> **Whether the patient has a prior medical history of coagulopathy (a blood clotting disorder).**

It indicates that the patient has been diagnosed with **a condition that affects the blood’s ability to clot properly before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of coagulopathy | 80,206 |
| **1** | Has coagulopathy history | 19,794 |

Total patients: **100,000**

Approximately:

- **80.2%** → No coagulopathy history  
- **19.8%** → Coagulopathy history  

---

### 3️ What is Coagulopathy

**Coagulopathy** refers to a group of conditions where the **blood clotting process is abnormal**.

Normally, when a person is injured, the body forms a **blood clot** to stop bleeding.  
In coagulopathy, this process may be **impaired or excessive**.

Types of clotting disorders include:

- **Bleeding disorders** (blood does not clot properly)
- **Hypercoagulable states** (blood clots too easily)

Examples of conditions that cause coagulopathy:

- hemophilia  
- liver disease  
- vitamin K deficiency  
- anticoagulant medication use (e.g., warfarin)  
- disseminated intravascular coagulation (DIC)

---

### 4️ Why This Feature Is Important in Emergency Triage

Coagulopathy is clinically important in emergency medicine because abnormal clotting can lead to **serious complications**, including:

- **severe bleeding**
- **internal hemorrhage**
- **stroke**
- **dangerous blood clots**
- **complications during trauma**

Patients with clotting disorders are at higher risk during situations such as:

- injuries  
- surgeries  
- infections  
- major trauma

Even minor injuries can result in **excessive bleeding**.

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider clotting disorders when evaluating symptoms such as:

- unusual bleeding  
- bruising  
- severe headaches  
- neurological symptoms  
- trauma-related injuries  

Example scenario:

Patient arrives with:

- head injury  
- persistent bleeding  
- abnormal vital signs  

If **hx_coagulopathy = 1**, clinicians may suspect:

- **internal bleeding**
- **increased hemorrhage risk**
- **clotting-related complications**

These situations may require **urgent medical evaluation and monitoring**.

---


This helps the model detect **patients with higher bleeding or clotting risk**.

---

### 6 Feature Type

`hx_coagulopathy` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Hematological risk indicator**

No scaling or normalization is required.

---

### 7 Role in the Triagegeist Model

This feature represents **baseline blood clotting vulnerability**.

Patients with coagulopathy may experience **severe bleeding or clotting complications during illness or injury**, which can significantly increase emergency severity.

Therefore, this variable helps improve the model’s ability to **identify high-risk patients with bleeding or clotting disorders**.


In [14]:
df_patient_history['hx_immunosuppressed'].value_counts()


hx_immunosuppressed
0    80108
1    19892
Name: count, dtype: int64

## Column: `hx_immunosuppressed`

### 1️ What This Column Means

- **hx** = history  
- **immunosuppressed** = weakened or suppressed immune system

This column represents:

> **Whether the patient has a prior medical history of being immunosuppressed.**

It indicates that the patient has a **weakened immune system before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | Not immunosuppressed | 80,108 |
| **1** | Immunosuppressed | 19,892 |

Total patients: **100,000**

Approximately:

- **80.1%** → Not immunosuppressed  
- **19.9%** → Immunosuppressed  

---

### 3️ What is Immunosuppression

**Immunosuppression** refers to a condition where the **immune system's ability to fight infections and diseases is reduced**.

This can occur due to:

- certain diseases (e.g., HIV, cancer)  
- chemotherapy treatments  
- organ transplant medications  
- autoimmune disease treatments  
- long-term steroid therapy  

When the immune system is suppressed, the body becomes **more vulnerable to infections and complications**.

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients who are immunosuppressed are considered **high-risk in emergency settings** because infections can:

- develop **more quickly**
- become **more severe**
- spread **rapidly throughout the body**

Common risks include:

- **sepsis (severe systemic infection)**
- **opportunistic infections**
- **delayed healing**
- **increased hospitalization risk**

Even minor infections can become **life-threatening** in immunosuppressed individuals.

---

### 5️ Clinical Importance During Triage

Emergency clinicians pay close attention to immunosuppression when patients present with symptoms such as:

- fever  
- fatigue  
- breathing difficulty  
- unexplained weakness  

Example scenario:

Patient arrives with:

- fever  
- rapid heart rate  
- weakness  

If **hx_immunosuppressed = 1**, clinicians may suspect:

- **serious infection**
- **sepsis risk**
- **compromised immune response**

These cases often require **rapid medical evaluation and treatment**.

---


This helps the model identify **patients with reduced immune defense**.

---

### 6 Feature Type

`hx_immunosuppressed` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Immune system vulnerability indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature helps represent **baseline immune system strength**.

Patients who are immunosuppressed have a **higher likelihood of developing severe infections and complications**, which can significantly affect emergency severity assessment.

Therefore, this variable improves the model’s ability to **detect high-risk infection-related emergencies**.


In [16]:
df_patient_history['hx_pregnant'].value_counts()


hx_pregnant
0    96427
1     3573
Name: count, dtype: int64

## Column: `hx_pregnant`

### 1️ What This Column Means

- **hx** = history  
- **pregnant** = currently pregnant or pregnancy-related status

This column represents:

> **Whether the patient is pregnant at the time of the emergency department visit.**

It indicates that the patient is **currently pregnant or has a pregnancy-related clinical status** when presenting to the emergency department.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | Not pregnant | 96,427 |
| **1** | Pregnant | 3,573 |

Total patients: **100,000**

Approximately:

- **96.4%** → Not pregnant  
- **3.6%** → Pregnant  

This distribution is expected because **pregnancy applies only to a small subset of the population**.

---

### 3️ What Pregnancy Means in Clinical Context

Pregnancy is a **physiological condition where a woman carries a developing fetus**. During pregnancy, the body undergoes major changes affecting:

- cardiovascular system  
- respiratory system  
- hormonal balance  
- blood circulation  
- immune function  

These changes can influence how diseases present and progress.

---

### 4️ Why This Feature Is Important in Emergency Triage

Pregnancy significantly affects **clinical decision making in emergency medicine** because both the **mother and fetus must be considered**.

Pregnant patients may present with pregnancy-related emergencies such as:

- **ectopic pregnancy**
- **preeclampsia**
- **placental abruption**
- **preterm labor**
- **severe bleeding**

Additionally, common illnesses may become **more dangerous during pregnancy**.

---

### 5️ Clinical Importance During Triage

Emergency clinicians immediately assess pregnancy status when patients present with symptoms such as:

- abdominal pain  
- vaginal bleeding  
- dizziness  
- severe headaches  
- high blood pressure  

Example scenario:

Patient arrives with:

- abdominal pain  
- dizziness  
- low blood pressure  

If **hx_pregnant = 1**, clinicians may suspect:

- **ectopic pregnancy**
- **pregnancy complications**
- **maternal or fetal risk**

These conditions may require **urgent obstetric evaluation**.

---


This helps the model recognize **pregnancy-related emergency risk**.

---

### 6 Feature Type

`hx_pregnant` is:

- **Binary feature (0 / 1)**
- **Medical status variable**
- **Obstetric risk indicator**

No scaling or normalization is required.

---

### 7 Role in the Triagegeist Model

This feature represents **pregnancy-related physiological and clinical risk**.

Pregnant patients may experience **unique complications that require specialized care and rapid triage decisions**, making pregnancy status an important factor in predicting emergency severity.

Therefore, this variable helps the model identify **potential obstetric emergencies and pregnancy-related complications**.


In [18]:
df_patient_history['hx_substance_use_disorder'].value_counts()


hx_substance_use_disorder
0    80057
1    19943
Name: count, dtype: int64

## Column: `hx_substance_use_disorder`

### 1️ What This Column Means

- **hx** = history  
- **substance_use_disorder** = problematic use of drugs or alcohol

This column represents:

> **Whether the patient has a prior medical history of substance use disorder (SUD).**

It indicates that the patient has previously been diagnosed with **a disorder involving misuse or dependence on substances such as alcohol, opioids, stimulants, or other drugs** before the current emergency department visit.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of substance use disorder | 80,057 |
| **1** | Has substance use disorder history | 19,943 |

Total patients: **100,000**

Approximately:

- **80.1%** → No substance use disorder history  
- **19.9%** → Substance use disorder history  

---

### 3️ What is Substance Use Disorder

**Substance Use Disorder (SUD)** is a medical condition characterized by **the repeated use of alcohol or drugs despite harmful consequences**.

It may involve substances such as:

- alcohol  
- opioids (e.g., heroin, prescription painkillers)  
- stimulants (e.g., cocaine, methamphetamine)  
- sedatives or benzodiazepines  
- other recreational or prescription drugs

SUD can affect both **physical health and mental health**.

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients with substance use disorder may present to the emergency department due to complications such as:

- **drug overdose**
- **withdrawal symptoms**
- **intoxication**
- **injuries related to substance use**
- **mental health crises**

Substance misuse can also worsen existing medical conditions and may affect **vital signs, behavior, and decision-making ability**.

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider substance use history when patients present with symptoms such as:

- altered consciousness  
- confusion  
- abnormal breathing  
- agitation or unusual behavior  

Example scenario:

Patient arrives with:

- reduced consciousness  
- slow breathing  
- abnormal vital signs  

If **hx_substance_use_disorder = 1**, clinicians may suspect:

- **drug overdose**
- **withdrawal syndrome**
- **substance-related complications**

These conditions may require **immediate medical intervention**.

---



### 6 Feature Type

`hx_substance_use_disorder` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Behavioral and clinical risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature helps represent **behavioral health risk and potential substance-related complications**.

Patients with substance use disorder may have **higher risk of overdose, withdrawal, injuries, and altered mental status**, all of which can significantly influence emergency triage severity.

Therefore, this variable helps improve the model’s ability to **identify high-risk substance-related emergency presentations**.


In [20]:
df_patient_history['hx_coronary_artery_disease'].value_counts()


hx_coronary_artery_disease
0    71818
1    28182
Name: count, dtype: int64

## Column: `hx_coronary_artery_disease`

### 1️ What This Column Means

- **hx** = history  
- **coronary_artery_disease (CAD)** = disease affecting the arteries that supply blood to the heart

This column represents:

> **Whether the patient has a prior medical history of coronary artery disease.**

It indicates that the patient has been previously diagnosed with **coronary artery disease before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of coronary artery disease | 71,818 |
| **1** | Has coronary artery disease history | 28,182 |

Total patients: **100,000**

Approximately:

- **71.8%** → No CAD history  
- **28.2%** → CAD history  

---

### 3️ What is Coronary Artery Disease

**Coronary Artery Disease (CAD)** occurs when the **coronary arteries that supply oxygen-rich blood to the heart become narrowed or blocked** due to plaque buildup.

This process is called **atherosclerosis**.

Plaque buildup reduces blood flow to the heart muscle and can cause serious cardiovascular problems.

Common symptoms include:

- chest pain (angina)  
- shortness of breath  
- fatigue  
- discomfort in the arms, neck, or jaw  

In severe cases, CAD can lead to **heart attacks**.

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients with coronary artery disease are considered **high-risk in emergency settings**, especially when presenting with symptoms such as:

- chest pain  
- shortness of breath  
- dizziness  
- abnormal heart rhythms  

CAD significantly increases the risk of:

- **acute myocardial infarction (heart attack)**
- **cardiac arrest**
- **severe arrhythmias**
- **heart failure**

Because of these risks, CAD history strongly influences **triage urgency**.

---

### 5️ Clinical Importance During Triage

Emergency clinicians immediately consider CAD history when evaluating symptoms such as:

- chest pain  
- pressure in the chest  
- shortness of breath  
- nausea or sweating  

Example scenario:

Patient arrives with:

- chest pain  
- sweating  
- rapid heart rate  

If **hx_coronary_artery_disease = 1**, clinicians may suspect:

- **heart attack**
- **unstable angina**
- **cardiac ischemia**

These conditions require **rapid cardiac evaluation and treatment**.

---


This helps the model identify **patients with underlying cardiovascular risk**.

---

### 6 Feature Type

`hx_coronary_artery_disease` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Cardiovascular risk indicator**

No scaling or normalization is required.

---

### 7 Role in the Triagegeist Model

This feature represents **baseline cardiovascular disease risk**.

Patients with coronary artery disease have a significantly higher likelihood of **cardiac emergencies**, which can rapidly become life-threatening.

Therefore, this variable helps improve the model’s ability to **predict severe cardiovascular-related emergency cases**.


In [22]:
df_patient_history['hx_stroke_prior'].value_counts()


hx_stroke_prior
0    72086
1    27914
Name: count, dtype: int64

## Column: `hx_stroke_prior`

### 1️ What This Column Means

- **hx** = history  
- **stroke_prior** = previous stroke event

This column represents:

> **Whether the patient has a prior history of stroke.**

It indicates that the patient has previously experienced **a stroke (cerebrovascular accident) before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No prior stroke history | 72,086 |
| **1** | History of prior stroke | 27,914 |

Total patients: **100,000**

Approximately:

- **72.1%** → No prior stroke  
- **27.9%** → Prior stroke history  

---

### 3️ What is a Stroke

A **stroke** occurs when the **blood supply to part of the brain is interrupted or reduced**, preventing brain tissue from receiving oxygen and nutrients.

There are two major types of stroke:

- **Ischemic stroke** – caused by a blood clot blocking a brain artery  
- **Hemorrhagic stroke** – caused by bleeding in the brain

Stroke can cause damage to brain cells within minutes.

Common symptoms include:

- sudden weakness on one side of the body  
- difficulty speaking or understanding speech  
- vision problems  
- severe headache  
- loss of balance or coordination  

---

### 4️ Why This Feature Is Important in Emergency Triage

Patients with a prior stroke are at **significantly higher risk of future neurological events**, including:

- **recurrent stroke**
- **transient ischemic attacks (TIAs)**
- **neurological deterioration**
- **vascular complications**

Stroke survivors often have **underlying vascular disease**, making them more vulnerable to further brain injury.

---

### 5️ Clinical Importance During Triage

Emergency clinicians pay close attention to stroke history when patients present with symptoms such as:

- weakness  
- confusion  
- speech difficulty  
- facial drooping  
- vision changes  

Example scenario:

Patient arrives with:

- sudden weakness in one arm  
- slurred speech  
- dizziness  

If **hx_stroke_prior = 1**, clinicians may suspect:

- **recurrent stroke**
- **transient ischemic attack (TIA)**
- **cerebrovascular complications**

These situations require **immediate neurological evaluation and imaging**.

---


This helps the model identify **patients at risk of neurological emergencies**.

---

### 6 Feature Type

`hx_stroke_prior` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Neurological and vascular risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature represents **baseline cerebrovascular vulnerability**.

Patients with a history of stroke have a **higher probability of recurrent neurological events**, which may require urgent medical intervention.

Therefore, this variable improves the model’s ability to **identify patients at risk of severe neurological emergencies**.


In [24]:
df_patient_history['hx_peripheral_vascular_disease'].value_counts()


hx_peripheral_vascular_disease
0    80261
1    19739
Name: count, dtype: int64

## Column: `hx_peripheral_vascular_disease`

### 1️ What This Column Means

- **hx** = history  
- **peripheral_vascular_disease (PVD)** = disease affecting blood vessels outside the heart and brain

This column represents:

> **Whether the patient has a prior medical history of peripheral vascular disease (PVD).**

It indicates that the patient has been previously diagnosed with **peripheral vascular disease before the current emergency department visit**.

---

### 2️ Data Distribution

| Value | Meaning | Count |
|------|--------|------|
| **0** | No history of peripheral vascular disease | 80,261 |
| **1** | Has peripheral vascular disease history | 19,739 |

Total patients: **100,000**

Approximately:

- **80.3%** → No PVD history  
- **19.7%** → PVD history  

---

### 3️ What is Peripheral Vascular Disease

**Peripheral Vascular Disease (PVD)** refers to conditions where **blood vessels outside the heart and brain become narrowed, blocked, or damaged**.

It commonly affects the **arteries supplying blood to the legs and arms**.

The most common cause is **atherosclerosis**, where fatty plaques build up inside arteries and restrict blood flow.

Common symptoms include:

- leg pain while walking (claudication)  
- numbness or weakness in limbs  
- coldness in lower legs or feet  
- slow-healing wounds on the legs or feet  

---

### 4️ Why This Feature Is Important in Emergency Triage

Peripheral vascular disease is important in emergency medicine because it often indicates **widespread vascular disease throughout the body**.

Patients with PVD have increased risk of:

- **heart attack**
- **stroke**
- **blood clots**
- **limb ischemia (reduced blood supply to limbs)**

Severe cases can lead to **critical limb ischemia**, where tissues begin to die due to lack of blood flow.

---

### 5️ Clinical Importance During Triage

Emergency clinicians consider PVD history when evaluating symptoms such as:

- leg pain or numbness  
- cold or pale limbs  
- wounds that do not heal  
- sudden loss of blood flow to a limb  

Example scenario:

Patient arrives with:

- severe leg pain  
- cold and pale foot  
- reduced pulse in the limb  

If **hx_peripheral_vascular_disease = 1**, clinicians may suspect:

- **acute limb ischemia**
- **vascular blockage**
- **circulatory complications**

These conditions require **urgent vascular assessment and treatment**.

---


This helps the model recognize **patients with underlying vascular disease**.

---

### 6 Feature Type

`hx_peripheral_vascular_disease` is:

- **Binary feature (0 / 1)**
- **Medical history variable**
- **Cardiovascular and vascular risk indicator**

No scaling or transformation is required.

---

### 7 Role in the Triagegeist Model

This feature represents **baseline vascular health and circulation risk**.

Patients with peripheral vascular disease often have **systemic atherosclerosis**, which increases the likelihood of serious cardiovascular or circulatory emergencies.

Therefore, this variable improves the model’s ability to **identify patients at risk of vascular-related emergency conditions**.


# ............................................................................The END...............................................................

np.int64(0)